spec/jobs/topic_reminder_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe Jobs::TopicReminder do
+  let(:admin) { Fabricate(:admin) }
+  let(:topic) { Fabricate(:topic, topic_status_updates: [
+    Fabricate(:topic_status_update, user: admin, status_type: TopicStatusUpdate.types[:reminder])
+  ]) }
+
+  before do
+    SiteSetting.queue_jobs = true
+  end
+
+  it "should be able to create a reminder" do
+    topic
+    topic_status_update = topic.topic_status_update(admin)
+    Timecop.freeze(1.day.from_now) do
+      expect {
+        described_class.new.execute(topic_status_update_id: topic_status_update.id)
+      }.to change { Notification.count }.by(1)
+      expect( admin.notifications.where(notification_type: Notification.types[:topic_reminder]).first&.topic_id ).to eq(topic.id)
+      expect( TopicStatusUpdate.where(id: topic_status_update.id).first ).to be_nil
```

Comment:
Maybe `expect(topic.reload.topic_status_updates).to eq([])` so that we ensure the topic does not have any topic status updates.